### PR TITLE
[NFC] Inject asSwiftSafeName as a dependency

### PR DESCRIPTION
**Pull request by czechboy0**
### Motivation

Arguably, we should have done this before, but here we go. In order to allow dynamically customizing the logic for converting OpenAPI to Swift identifiers (in support of staging SOAR-0001 gradually), we need to dependency-inject the `(String) -> String` function to all the places that need to do this conversion.

### Modifications

This PR doesn't change any behavior (NFC - no functional change), it just refactors the code to have a single funnel point through which all the conversion goes, which has access to the `Config` and can inspect feature flags.

Right now, the default behavior is still the same, but this PR creates room for adding the alternative logic from SOAR-0001.

### Result

NFC, prepares room for a subsequent PR.

Now it's possible to conditionalize the Swift name computing logic based on feature flags.

### Test Plan

All tests are passing, adapted as needed, created conveniences in tests for getting a default `Config`, but also made it possible to test the logic by explicitly passing the feature flag.

### CODE REVIEW

**Comment by czechboy0** [comment]
FYI, @denil-ct, once this lands, you'll be able to add your logic [here](https://github.com/apple/swift-openapi-generator/pull/148/files#diff-f2d822fd23049dd135cec20445850f1f545813cc79cb0b121dbbbeda01d338c4R76-R79) and then test is by providing the feature flag `proposal0001`.

**Review by simonjbeaumont — COMMENTED** [review]
I don't think we should be mixing in the injected function with the description types.

Either the names we compute within them should be safed by the thing that constructs them, or the computed properties should be made functions that take a safer, or they should return unsafed names, which the caller then safes.

I think in this way, we might preserve some better layering? WDYT?

File: Sources/_OpenAPIGeneratorCore/Extensions/String.swift [hunk 1]
```diff
@@ -61,7 +32,7 @@ fileprivate extension String {
     ///
     /// In addition to replacing illegal characters with an underscores, also
     /// ensures that the identifier starts with a letter and not a number.
-    var sanitizedForSwiftCode: String {
+    var safeForSwiftCode: String {
```
[the remark this answers is not in the document]
Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Extensions/String.swift:35]: Will do.

File: Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/SwiftSafeNames.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//
+//===----------------------------------------------------------------------===//
+import Foundation
+
+extension FileTranslator {
+
+    /// Returns a copy of the string modified to be a valid Swift identifier.
+    ///
+    /// - Parameter string: The string to convert to be safe for Swift.
+    func swiftSafeName(for string: String) -> String {
```
[the remark this answers is not in the document]
Comment (czechboy0) [reply Sources/_OpenAPIGeneratorCore/Translator/CommonTranslations/SwiftSafeNames.swift:21]: `swiftSafeName` being a _method_ of `FileTranslator`, you can grab it with:

```swift
let foo: (String) -> String = translator.swiftSafeName
```

This captures the function which captures the config value.

Or would you prefer something like this?

```swift
var swiftSafeName: (String) -> String {
  { string in
    // implementation that switches based on config
  }
}
```

**Review by simonjbeaumont — APPROVED** [review]

**Pull request merged**